lib/gringotts/gateways/mercadopago.ex

```diff
@@ -0,0 +1,376 @@
+  ```
+  iex> card = %{first_name: "John",
+                last_name: "Doe",
+                number: "4509953566233704",
+                year: 2099,
+                month: 12,
+                verification_code: "123",
+                brand: "VISA"}
+  ```
+
+  We'll be using these in the examples below.
+
+  [gs]: https://github.com/aviabird/gringotts/wiki/
+  [home]: https://www.mercadopago.com
+  [example]: https://github.com/aviabird/gringotts_example
+  """
+
+  # The Base module has the (abstract) public API, and some utility
+  # implementations.  
+  @base_url "https://api.mercadopago.com"
+  use Gringotts.Gateways.Base
+  alias Gringotts.CreditCard
+  # The Adapter module provides the `validate_config/1`
+  # Add the keys that must be present in the Application config in the
+  # `required_config` list
+  use Gringotts.Adapter, required_config: [:public_key, :access_token]
+  
+  import Poison, only: [decode: 1]
+
+  alias Gringotts.{CreditCard,
+                   Response}
+
+  @doc """
+  Performs a (pre) Authorize operation.
+
+  The authorization validates the `card` details with the banking network,
+  places a hold on the transaction `amount` in the customer’s issuing bank.
+
+  MERCADOPAGO's `authorize` returns a map containing authorization ID string which can be used to:
+
+  * `capture/3` _an_ amount.
+  * `void/2` a pre-authorization.
+  
+  ## Note
+
+  > If there's anything noteworthy about this operation, it comes here.
+
+  ## Example
+      iex> amount = Money.new(42, :BRL)
+      iex> card = %Gringotts.CreditCard{
+            first_name: "John",
+            last_name: "Doe",
+            number: "4509953566233704",
+            year: 2099,
+            month: 12,
+            verification_code: "123",
+            brand: "VISA"
+        }
+      iex> opts = [email: "xyz@test.com",
+           order_id: 123125,
+           customer_id: "308537342-HStv9cJCgK0dWU",
+           payment_method_id: "visa",
+           config: %{access_token: "TEST-2774702803649645-031303-1b9d3d63acb57cdad3458d386eee62bd-307592510",
+           public_key: "TEST-911f45a1-0560-4c16-915e-a8833830b29a"}]
+      iex> {:ok, auth_result} = Gringotts.authorize(Gringotts.Gateways.Mercadopago, amount, card, opts)
+      iex> auth_result.id # This is the authorization ID
+      iex> auth_result.token # This is the customer ID/token
+
+  """
+
+  @spec authorize(Money.t(), CreditCard.t(), keyword) :: {:ok | :error, Response}
+  def authorize(amount , %CreditCard{} = card, opts) do
+    # commit(args, ...)
+    {_, value, _, _} = Money.to_integer_exp(amount)
+    if(is_nil(opts[:customer_id])) do
+      customer_id = get_customer_id(opts)
+      opts = opts ++ [customer_id: customer_id]
+    end
+    token_id = get_token_id(card, opts)
+    opts = opts ++ [token_id: token_id]
+
+    body = get_authorize_body(value, card, opts, opts[:token_id], opts[:customer_id]) |> Poison.encode!()
```

Comment:
If you are sending opts as an argument, what is the need to send opts[:token_id] and opts[:customer_id] separately along with it?